### PR TITLE
Update CONTRIBUTION.md with pdm->poetry and pytest instructions

### PR DESCRIPTION
**Pull request by matt-bernstein**

### CODE REVIEW

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/6e420f3edc73eebd7ddcd71eb903d099283a8d5a is created.

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -67,47 +67,47 @@ Contributions that align with the items detailed in our roadmap, found in the ma
...
 ```
 
-You can expand with additional test dependencies using `pdm` with
-
+Adala uses [pytest-recording](https://pypi.org/project/pytest-recording/) to cache network calls in [VCR cassette files](https://vcrpy.readthedocs.io/en/latest/), so that the test suite can be run in CI and without network access. When adding tests that make network calls, such as to OpenAI, run the test suite with
 ```bash
-pdm add -dG test <package-name>
+pytest --record_mode=once
 ```
+and commit the resulting cassette `.yml` files that are generated. To modify existing tests that make network calls, delete their cassette files before running pytest.
```
Comment (niklub) [inline CONTRIBUTION.md:94]: let's add the command for rerunning the tests with `--block-network` option

**Review by niklub — APPROVED** [review]
lgtm

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/63611ad971715b5cd0d6b5572ca52f1352416d1c is created.

**Pull request merged**